app/assets/javascripts/admin/modules/document-history.js

```diff
@@ -0,0 +1,45 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  function DocumentHistory (module) {
+    this.module = module
+  }
+
+  DocumentHistory.prototype.init = function () {
+    this.setupEventListeners()
+  }
+
+  DocumentHistory.prototype.setupEventListeners = function () {
+    var documentHistoryTab = document.querySelector('#history_tab')
```

Comment:
```suggestion
    var documentHistoryTab = this.module.querySelector('#history_tab')
```

Reply:
So unfortunately the module actually only returns the link to the tab not the tab itself which are both children of the govuk-tabs div.

I can do this to avoid using the document if it's better?   

` var documentHistoryTab = this.module.closest('.govuk-tabs').querySelector('#history_tab')` 

wdyt?

Reply:
Sorry for the drive by comment, saw it pop up in my email.

I imagine that if the elements you are interacting with are outside the module you probably want to move the module to contain it. But I guess you don't need that as I think you're just listening for clicks on elements within the tab? so you could probably event listeners on the newer/older links

Reply:
with the current implementation the whole tab gets replaced including the links. So adding them to the parent elements seems like the easiest solution.

We could go down the route of only adding in the section below where the links are rendered by munging the response, but this seemed simpler. Wdyt?

Reply:
Ah sure, that makes sense. I think in that case you do have a bit of an isolation issue as the replacement also replaces the element that generated the module - as that'd no longer be in the DOM after a page change. This isn't a cardinal sin but does feel like a bit of a smell as there isn't encapsulation

I think you'd do best to create an element inside the history which is a wrapper for this i.e the tab contents contain:

```
<div data-module="edition-history-paginator">
  <%= history_with_pagination %>
</div>
```

Then that div can be the module and can have it's contents replaced while still persisting with each pagination and only interacting with elements inside of it.